### PR TITLE
[PO-12] - Model SlotInEpoch

### PR DESCRIPTION
**Pull request by mirkoAlic**

### CODE REVIEW

File: src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package io.iohk.praos.domain
+
+class SlotInEpochCalculator(epochLength: Int, slotDurationInMilliseconds: Int) {
```
Comment (ntallar) [inline src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala]: Minor detail: Maybe this class could be a `case class` as it isn't mutable?

File: src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  def calculateSlotNumber(timeProvider: TimeProvider): Int =
+    ((timeProvider.getTime - timeProvider.initialTime) / slotDurationInMilliseconds) + 1
+
+  def calculateEpochNumber(timeProvider: TimeProvider): Int =
```
Comment (ntallar) [inline src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala]: Minor detail: Maybe we could add here and in `calculateSlotNumber` a comment regarding these numbers starting from 1 and not 0?

Comment (mirkoAlic) [reply src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala]: :+1:

File: src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala (hunk 1 above)
Comment (mirkoAlic) [reply src/main/scala/io/iohk/praos/domain/SlotInEpochCalculator.scala]: :+1:

**Review by ntallar — APPROVED** [review]
LGTM!

**Pull request merged**